common/app/assets/DiscussionAssets.scala

```diff
@@ -0,0 +1,99 @@
+package assets
+
+import java.net.URI
+
+import app.LifecycleComponent
+import common.{AkkaAgent, ExecutionContexts, GuardianConfiguration, JobScheduler, Logging}
+import conf.switches.Switches
+import play.api.libs.ws.{WSClient, WSResponse}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+/**
+  * External assets are not hosted by frontend but referenced in an assets map
+  *
+  * This class pull the assets map regularly and translates a generic name into a full URL
+  *
+  * The map is a JSON object looking like
+  * {
+  *   "name": "js/name.min.hash.js"
+  * }
+  * The path in the object value is relative to the assets map
+  */
+class DiscussionExternalAssetsLifecycle (config: GuardianConfiguration, wsClient: WSClient, jobs: JobScheduler)
+  extends LifecycleComponent with ExecutionContexts with Logging {
+
+  def refresh(): Future[Unit] = {
+    config.discussion.frontendAssetsMap match {
+      case Some(url) if Switches.DiscussionFetchExternalAssets.isSwitchedOn =>
+        fetchAssetsMap(url, wsClient).map(
```

Comment:
Yup, that's the whole point. I want to be able to deploy discussion independently of frontend.

Think of discussion as an interactive / embded / atom / whatever with its own lifecycle and needs